packages/collection/src/collection.ts

```diff
@@ -669,52 +669,93 @@ export class Collection<K, V> extends Map<K, V> {
 	}
 
 	/**
-	 * The intersect method returns a new structure containing items where the keys and values are present in both original structures.
+	 * The intersection method returns a new collection containing the items where the key is present in both collections.
 	 *
 	 * @param other - The other Collection to filter against
+	 * @example
+	 * ```ts
+	 * const col1 = new Collection([['a', 1], ['b', 2]]);
+	 * const col2 = new Collection([['a', 1], ['c', 3]]);
+	 * const intersection = col1.intersection(col2);
+	 * console.log(col1.intersection(col2));
+	 * // => Collection { 'a' => 1 }
+	 * ```
 	 */
-	public intersect<T>(other: ReadonlyCollection<K, T>): Collection<K, T> {
-		const coll = new this.constructor[Symbol.species]<K, T>();
-		for (const [k, v] of other) {
-			if (this.has(k) && Object.is(v, this.get(k))) {
-				coll.set(k, v);
-			}
+	public intersection<T>(other: ReadonlyCollection<K, T>): Collection<K, T | V> {
+		const coll = new this.constructor[Symbol.species]<K, T | V>();
+
+		for (const [k, v] of this) {
+			if (other.has(k)) coll.set(k, v);
 		}
 
 		return coll;
 	}
 
 	/**
-	 * The subtract method returns a new structure containing items where the keys and values of the original structure are not present in the other.
+	 * The union method returns a new collection containing the items where the key is present in either of the collections.
```

Comment:
At first I would suggest this to be more direct and consistent with my other suggestions, since this is the mathematical definition...
```suggestion
	 * Returns a new collection containing the items where the key is present in either of the collections.
```

But I don't feel like it's right, take the example:

```typescript
const col1 = new Collection([['a', 1], ['b', 2]]);
const col2 = new Collection([['a', 1], ['c', 3]]);
const union = col1.union(col2);
console.log(union);
// => Collection { 'a' => 1, 'b' => 2, 'c' => 3 }
```

This can be a given, but once we change things a little, it behaves in a way that is inconsistent with the other methods, namely, the fact that `B`'s values have precedence over `A`'s.

```typescript
const col1 = new Collection([['a', 1], ['b', 2]]);
const col2 = new Collection([['a', 4], ['c', 3]]);
const union = col1.union(col2);
console.log(union);
// => Collection { 'a' => 4, 'b' => 2, 'c' => 3 }
//                        ^ union.get('a') changed!
```

I believe this behaviour would suit more for a "merge" — which has its own [proposal](https://github.com/tc39/proposal-collection-methods) with a [polyfill](https://github.com/tc39/proposal-collection-methods/blob/99480f94980a69efd1f24bdd7381f571d688f53d/map-polyfill.js#L88-L92) — rather than an "union". In my head, the way to replicate an union behaviour is to skip setting B's values if their keys are already present, as shown by the [`Set.prototype.union` proposal](https://github.com/tc39/proposal-set-methods/blob/b67c7a22f713ac70e5fa7d596aa85490fad17d7e/spec/index.html#L25-L26).